pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/startree/AggregationFunctionColumnPair.java

```diff
@@ -18,26 +18,61 @@
  */
 package org.apache.pinot.segment.spi.index.startree;
 
+import java.util.Arrays;
 import java.util.Comparator;
+import java.util.Objects;
+import org.apache.commons.configuration.Configuration;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.apache.pinot.segment.spi.compression.ChunkCompressionType;
+import org.apache.pinot.spi.config.table.StarTreeAggregationConfig;
 
 
 public class AggregationFunctionColumnPair implements Comparable<AggregationFunctionColumnPair> {
   public static final String DELIMITER = "__";
   public static final String STAR = "*";
+
   public static final AggregationFunctionColumnPair COUNT_STAR =
       new AggregationFunctionColumnPair(AggregationFunctionType.COUNT, STAR);
 
   private final AggregationFunctionType _functionType;
   private final String _column;
+  private final ChunkCompressionType _chunkCompressionType;
 
-  public AggregationFunctionColumnPair(AggregationFunctionType functionType, String column) {
+  public AggregationFunctionColumnPair(AggregationFunctionType functionType, String column,
+      ChunkCompressionType compressionType) {
     _functionType = functionType;
     if (functionType == AggregationFunctionType.COUNT) {
       _column = STAR;
     } else {
       _column = column;
     }
+    _chunkCompressionType = compressionType;
+  }
+
+  public AggregationFunctionColumnPair(AggregationFunctionType functionType, String column) {
+    this(functionType, column, ChunkCompressionType.PASS_THROUGH);
+  }
+
+  public static AggregationFunctionColumnPair fromConfiguration(Configuration metadataProperties) {
+    if (metadataProperties == null) {
+      return null;
+    }
+    AggregationFunctionType functionType = AggregationFunctionType.getAggregationFunctionType(
+        metadataProperties.getString(StarTreeV2Constants.MetadataKey.FUNCTION_TYPE));
+    String columnName = metadataProperties.getString(StarTreeV2Constants.MetadataKey.COLUMN_NAME);
+    ChunkCompressionType compressionType =
+        ChunkCompressionType.valueOf(metadataProperties.getString(StarTreeV2Constants.MetadataKey.COMPRESSION_TYPE));
+    return new AggregationFunctionColumnPair(functionType, columnName, compressionType);
+  }
+
+  // Adds current object to Configuration by reference
+  public void addToConfiguration(Configuration configuration) {
+    String configPrefix = StarTreeV2Constants.MetadataKey.AGGREGATION_CONFIG + "." + this.toColumnName();
```

Comment:
Suggest not using column name as the prefix because in the future we might want to deprecate it. Using index might be good: `aggregate.0.function`, `aggregate.0.column`, `aggregate.0.codec`